larpandoracontent/LArControlFlow/StitchingCosmicRayMergingTool.cc

```diff
@@ -538,7 +581,16 @@ void StitchingCosmicRayMergingTool::StitchPfos(const MasterAlgorithm *const pAlg
             }
 
             for (const ParticleFlowObject *const pPfoToShift : pfoVector)
-                pAlgorithm->ShiftPfoHierarchy(pPfoToShift, pfoToLArTPCMap, x0);
+            {
+                if (pfoToLArTPCMap.find(pPfoToShift) != pfoToLArTPCMap.end())
+                {
+                    const LArTPC *const pActiveLArTPC(pfoToLArTPCMap.at(pPfoToShift));
+                    const LArTPC *const pPartnerLArTPC(stitchedLArTPCs.first == pActiveLArTPC ? stitchedLArTPCs.second : stitchedLArTPCs.first);
+                    const float sign(pActiveLArTPC->GetCenterX() < pPartnerLArTPC->GetCenterX() ? 1.f : -1.f);
```

Comment:
Just looking for reassurance that you've checked this through, and thought about e.g. any cosmic ray reconstruction (rare, but required) that might need to happen in DUNE FD, with different APA, CPA structure.

Isn't it possible that you might want to move both tracks (in both TPCs) in the opposite sense in order to achieve a good stitch (so cosmic ray arrived after Tbeam, rather than before Tbeam)?

Or is there something somewhere else that precludes this? @astblake?

Reply:
Hi John,
Thank you for the comments.  They were really useful and I've implemented them.  I think what I didn't particularly like was the try-catch block structure as I felt it was too big, but by moving the try block logic to a separate function it looks quite clean now.  

I think you might be right about there being an issue if the CPA/APA structure is different as this logic assumes that the 3D particle appears in the 3D drift volume it was measured in, when that might not be the case.  I will have more of a think about this and look at a few more events to check everything still makes sense.  Especially for DUNEFD too.  

Cheers
Steve